app/hotels/src/allHotels/AllHotelsSearchList.js

```diff
@@ -42,47 +106,91 @@ export class AllHotelsSearchList extends React.Component<Props> {
       );
     } else {
       return (
-        <ScrollView style={{ paddingVertical: 3 }}>
-          {hotels.map(edge => {
-            if (edge) {
-              const { node: hotel } = edge;
-              return (
-                <AllHotelsSearchRow
-                  data={hotel}
-                  openSingleHotel={this.props.openSingleHotel}
-                  key={hotel && hotel.id}
-                />
-              );
-            }
-          })}
-        </ScrollView>
+        <View style={styles.container}>
+          <FlatList
+            data={hotels}
+            renderItem={this.renderListItem}
+            keyExtractor={this.keyExtractor}
+            onEndReached={this.loadMore}
+          />
+          {this.renderLoadingIndication()}
+        </View>
       );
     }
   };
 }
 
-const mapDispatchToProps = dispatch => ({
+const actions = dispatch => ({
   setCurrentSearchStats: (currentSearchStats: CurrentSearchStats) =>
     dispatch({
       type: 'setCurrentSearchStats',
       currentSearchStats,
     }),
 });
 
-export default createFragmentContainer(
-  connect(null, mapDispatchToProps)(AllHotelsSearchList),
-  graphql`
-    fragment AllHotelsSearchList on HotelAvailabilityConnection {
-      edges {
-        node {
-          id
-          ...AllHotelsSearchRow
+export default createPaginationContainer(
+  connect(null, actions)(AllHotelsSearchList),
+  {
+    data: graphql`
+      fragment AllHotelsSearchList_data on RootQuery {
+        allAvailableHotels(
+          search: $search
+          filter: $filter
+          options: $options
+          first: $first
+          after: $after
+        ) @connection(key: "AllHotels_allAvailableHotels") {
+          pageInfo {
+            hasNextPage
+            hasPreviousPage
+            startCursor
+            endCursor
+          }
+          edges {
+            node {
+              id
+              ...AllHotelsSearchRow
+            }
+          }
+          stats {
+            priceMax
+            priceMin
+          }
         }
       }
-      stats {
-        priceMax
-        priceMin
+    `,
+  },
+  {
+    direction: 'forward',
+    getConnectionFromProps(props) {
+      return props.data && props.data.allAvailableHotels;
+    },
+    getFragmentVariables(prevVars, totalCount) {
+      return {
+        ...prevVars,
+        count: totalCount,
+      };
+    },
```

Comment:
Options `direction`, `getConnectionFromProps` and `getFragmentVariables` are optional and I think unnecessary in this case. What do you think about removing them?

Reply:
Nice, I did not know they were optional, I will remove them